### PR TITLE
feat: documentation for new workflow config options

### PR DESCRIPTION
**Pull request by kirederik**
* plus: reorder the reference sidebar

### CODE REVIEW

File: docs/main/05-reference/06-workflows.md [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+
+Pipeline containers can control aspects of how Kratix behaves by creating
+special files in this directory:
+
+- `destination-selectors.yaml` can be added to any Promise to
+  further refine where the resources in `/kratix/output` will be
+  [scheduled](./multicluster-management).
+- `status.yaml` allows the Pipeline to communicate information about the
+  Resource back to the requester. See [status documentation
+  for more information](./resources/status).
```
Comment (aclevername) [inline docs/main/05-reference/06-workflows.md:139]: not for Promises I think?

File: docs/main/05-reference/06-workflows.md [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+
+Kratix will set the following environment variables for all containers in the
+workflow:
+
+
+| Variable                  | Description     |
+| ------------------------- | --------------- |
+| `KRATIX_WORKFLOW_ACTION`  | The action that triggered the workflow. Either `configure` or `delete`. |
+| `KRATIX_WORKFLOW_TYPE`    | The type of workflow. Either `resource` or `promise`. |
+| `KRATIX_PROMISE_NAME`     | The name of the Promise. |
```
Comment (aclevername) [inline docs/main/05-reference/06-workflows.md:155]: I think its `KRATIX_PROMISE`

Comment (aclevername) [reply docs/main/05-reference/06-workflows.md:155]: maybe it should be `KRATIX_PROMISE_NAME` though. FWIW I jsut added this last friday to help my research day project lol

Comment (kirederik) [reply docs/main/05-reference/06-workflows.md:155]: i started writing `KRATIX_PR` and copilot autocompleted. i confess i didn't check 🙈

File: docs/main/05-reference/06-workflows.md (hunk 1 above)
Comment (aclevername) [reply docs/main/05-reference/06-workflows.md:139]: bug driven development, i like it

File: docs/main/05-reference/06-workflows.md (hunk 2 above)
Comment (kirederik) [reply docs/main/05-reference/06-workflows.md:155]: i think kratix_promise_name is better btw

Comment (aclevername) [reply docs/main/05-reference/06-workflows.md:155]: go for it, i can update my research day work

File: docs/main/05-reference/06-workflows.md [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+  volumes: # optional
+  containers:
+    - name: # container name
+      image: # container image
+      # optional fields
+      command: []
+      args: []
+      env: []
+      envFrom: []
+      volumeMounts: []
```
[the remark this answers is not in the document]
Comment (kirederik) [reply docs/main/05-reference/06-workflows.md:77]: maybe... but maybe not :p

**Pull request merged**